File: HW07/131044061_CSE241_HW07_caner_karakas/hw07/src/Rectangle.java


import javax.swing.JFrame;
import javax.swing.JPanel;
import java.awt.Color;
import java.awt.Component;
import java.awt.Graphics;

public class Rectangle extends JPanel implements Shape {
	
	private static final long serialVersionUID = 1L;
	private static int numOfShape = 0;
	private double width;
	private double height;
	private double pos_x;
	private double pos_y;
	
	Rectangle(){}
	/**
	 * take Rectangle object
	 * @param rect
	 * @throws Exception
	 */
	Rectangle(Rectangle rect) throws Exception{
		this.set_width(rect.get_width());
		this.set_height(rect.get_height());
		this.set_coords(rect.get_pos_x(), rect.get_pos_y());
	}
	/**
	 * take double params
	 * @param w
	 * @param h
	 * @throws Exception
	 */
	Rectangle(double w, double h) throws Exception{
		this.set_width(w);
		this.set_height(h);
	}
	

	@Override
	public int compareTo(Shape other) {
		if(this.area() == other.area())
			return 1;
		return 0;
	}

	@Override
	public double area() {return width * height;}
	

	@Override
	public double perimeter() {return 2 * width * height;}

	@Override
	public Shape increment() {
		this.pos_x++;
		this.pos_y++;
		return this;
	}

	@Override
	public Shape decrement() {
		this.pos_x--;
		this.pos_y--;
		return this;
	}

	@Override
	public void draw(Graphics graph) {
		this.paintComponent(graph);
	}
	
	@Override
	public void paintComponent(Graphics graph){
		super.paintComponent(graph);
		graph.setColor(Color.GREEN);
		graph.drawRect((int)this.get_pos_x(), (int)this.get_pos_y(), (int)this.get_width(), (int)this.get_height());
	}
	/**
	 * set_width
	 * @param w
	 * @throws Exception
	 */
	public void set_width(double w) throws Exception{
		if(w>0){
			width = w;
		}
		else{
			new Exception ();
			/*System.out.printf("Error!!");
			System.exit(0);*/
		}
	}
	/**
	 * set_height
	 * @param h
	 * @throws Exception
	 */
	public void set_height(double h) throws Exception{
		if(h>0){
			height = h;
		}
		else{
			new Exception ();
			/*System.out.printf("Error!!");
			System.exit(0);*/
		}
	}
	
	public double get_width()	{return width;}
	
	public double get_height() {return height;}
	/**
	 * set_pos_x
	 * @param x
	 * @throws Exception
	 */
	public void set_pos_x(double x) throws Exception{
		if(x>=0){
			pos_x = x;
		}
		else{/*
			System.out.printf("Error!!");
			System.exit(0);*/
			new Exception ();
		}
	}
	/**
	 * set_pos_y
	 * @param y
	 * @throws Exception
	 */
	public void set_pos_y(double y) throws Exception{
		if(y>=0){
			pos_y = y;
		}
		else{
			new Exception ();
			/*System.out.printf("Error!!");
			System.exit(0);*/
		}
	}
	
	public double get_pos_x(){return pos_x;}
	
	public double get_pos_y() {return pos_y;}
	
	static int get_numOFShape() {numOfShape++;return numOfShape;}
	
	static int reset_numOfShape(){numOfShape = 0;return numOfShape;}
	
	static double result_area(Rectangle rect){return numOfShape * rect.area();}
	/**
	 * set_coords
	 * @param x
	 * @param y
	 * @throws Exception
	 */
	public void set_coords(double x, double y) throws Exception {
		set_pos_x(x);
		set_pos_y(y);
	}
	/**
	 * set_lines
	 * @param w
	 * @param h
	 * @throws Exception
	 */
	public void set_lines(double w, double h) throws Exception {
		this.set_width(w);
		this.set_height(h);
	}

}

File: HW03/131044061_CSE241_HW3/Rectangle.cpp
#include<string>
#include<fstream>
#include <iostream>
#include "Rectangle.h"
using namespace std;

Rectangle::Rectangle() {
	width = 0;
	height = 0;
	pos_x = 0;
	pos_y = 0;
}

void Rectangle::set_lines(int w, int h) {
	if(w>1 && h>1){
		width = w;
		height = h;
	}
	else{
		cout<<"rect set etmedi"<<endl;
	}
}

int Rectangle::get_width() const {
	return width;
}

int Rectangle::get_height() const {
	return height;
}

void Rectangle::set_coords(double p_x, double p_y) {
	if(p_x>=0.0 && p_y>=0.0){
		pos_x = p_x;
		pos_y = p_y;
	}
	else{
		cout<<"rect koordinantlar set olmadi"<< endl;
	}
}

double Rectangle::get_pos_x() const {
	return pos_x;
}

double Rectangle::get_pos_y() const {
	return pos_y;
}

const double Rectangle::area() {
	return (double)get_width() * (double)get_height();
}

const double Rectangle::lenght() {
	return 2*((double)get_width() + (double)get_height());
}


const Rectangle Rectangle::operator ++() {
	this->pos_x++;
	this->pos_y++;
	this->set_coords(pos_x, pos_y);
	Rectangle temp;
	temp.set_coords(pos_x, pos_y);
	return temp;
}

const Rectangle Rectangle::operator ++(int pre) {
	++this->pos_x;
	++this->pos_y;
	this->set_coords(pos_x, pos_y);
	Rectangle temp;
	temp.set_coords(pos_x, pos_y);
	return temp;
}

const Rectangle Rectangle::operator --() {
	this->pos_x--;
	this->pos_y--;
	this->set_coords(pos_x, pos_y);
	Rectangle temp;
	temp.set_coords(pos_x, pos_y);
	return temp;
}

const Rectangle Rectangle::operator --(int pre) {
	--this->pos_x;
	--this->pos_y;
	this->set_coords(pos_x, pos_y);
	Rectangle temp;
	temp.set_coords(pos_x, pos_y);
	return temp;
}

const bool Rectangle::operator ==(Rectangle& other) {
	return this->area() == other.area();
}

const bool Rectangle::operator !=(Rectangle& other) {
	return !(this->area() == other.area());
}

ofstream& operator <<(ofstream& ofs, const Rectangle& rect) {
	ofs<<rect.get_idntfy()<<" "<<rect.get_width_word()<<rect.get_width()
			<<"' "<<rect.get_height_word()<<rect.get_height()<<"' "<<rect.get_coord_x()
			<<rect.get_pos_x()<<"' "<<rect.get_coord_y()<<rect.get_pos_y()<<"' "
			<<rect.get_stroke()<<rect.fill<<endl;
	return ofs;
}

string Rectangle::get_idntfy() const {
	return idntfy;
}

string Rectangle::get_coord_x() const {
	return coord_x;
}

string Rectangle::get_coord_y() const {
	return coord_y;
}

string Rectangle::get_width_word() const {
	return width_word;
}

string Rectangle::get_height_word() const {
	return height_word;
}

string Rectangle::get_stroke() const {
	return stroke;
}

void Rectangle::draw(ofstream & ofs) const {
	ofs<<*this;
}

const Rectangle Rectangle::operator +(double num) {
	Rectangle temp;
	temp.set_lines((this->get_width()+num), (this->get_height()+num));
	return temp;
}

const Rectangle Rectangle::operator -(double num) {
	Rectangle temp;
	temp.set_lines((this->get_width()-num), (this->get_height()-num));
	return temp;
}

int Rectangle::get_numOFShape() {
	numOfShape++;
	return numOfShape;
}

double Rectangle::result_area(Rectangle & rect) {
	return numOfShape * rect.area();
}







